### PR TITLE
Ember 4.0 release blog post

### LINKED ISSUE #1013: Ember 4 release blog post tracking
See [The road to Ember 4](https://blog.emberjs.com/the-road-to-ember-4-0) for timelines and some features to cover.

This ticket will likely be worked on by the framework core team.

Additional topics:
- We should push the Super Rentals tutorial again with the release post. Devs could ignore Octane and therefore may have missed these resources in our Octane announcements. They will need them now. Godfrey suggests framing it like "If you need to catch up or refresh your knowledge on the latest and greatest, check out the tutorial"

## Steps to Reproduce

1.
2.
3.

### PR DESCRIPTION
**Pull request by jenweber**
This is the starting place for the blog post. Content needs to be filled in by other team members.

**Sign-off:**
- [x] ember-data (@mixonic)
- [x] ember-cli (@mixonic)
- [x] ember.js (@mixonic)
- [x] learning team (@mixonic)

**When to merge**
- After all teams have reviewed/signed off
- After Guides & API docs for v4 are live. Please @ the learning team.

Closes https://github.com/ember-learn/ember-blog/issues/1013

### CODE REVIEW

**Comment by jenweber** [comment]
@mixonic do you have anything in progress that you can push up? Feel free to add to this branch directly.

**Comment by jenweber** [comment]
@kategengler @rwjblue I just pushed up a very rough draft. Please review for accuracy and general organization. I am especially looking for feedback about whether to include the major version soapbox. I can edit to to make it shorter. I'm thinking of writing a separate post about release strategy, and would move the bulk of it there. Thank you!

**Comment by jenweber** [comment]
My to-dos:
- fix typos
- Make spellchecker happy
- Add a section with where to get help
- Link to Octane

File: content/ember-4-0-released.md [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+  - '2021'
+  - version-4-x
+---
+
+
+Today, the Ember project is releasing version 4.0.0 of Ember.js, Ember Data, and
+Ember CLI. Ember it focuses
+the framework by removing long-deprecated APIs and support for legacy platforms.
+With these changes, Ember Octane becomes the norm!
+As planned, this release introduces no new features.
```
Comment (NullVoxPopuli) [inline content/ember-4-0-released.md]: This is orthogonal to the release post, but I feel like we need a subclass of semver to communicate our sort of take on this. Some projects just do breaking changes whenever they feel like and never deprecate things, and they burn through majors like it's their job. :thinking: Maybe @wycats' wordsmithery has some ideas <3

Comment (NullVoxPopuli) [reply content/ember-4-0-released.md]: (though, I now see this is expanded on in the next section, but good words (ie: a name) give good marketing, so :shrug: )

File: content/ember-4-0-released.md [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+## Trying Ember for the first time
+
+If you're interested in trying Ember for the
+first time today, get started by running:
+
+```bash
+npx ember-cli new my-project
+cd my-project
+npx ember serve # Then visit http://localhost:4200
```
Comment (NullVoxPopuli) [inline content/ember-4-0-released.md]: does `npx ember` point at ember-cli?

File: content/ember-4-0-released.md (hunk 1 above)
Comment (jenweber) [reply content/ember-4-0-released.md]: That's a great point. I think we need a full blog post on this topic. We just need a name for it. I don't think there is one.

File: content/ember-4-0-released.md (hunk 2 above)
Comment (jenweber) [reply content/ember-4-0-released.md]: Not exactly? This is invalid:

```
npx ember new my-test-app
npm ERR! could not determine executable to run
```

This _is_ valid.
```
npx ember-cli serve
```

Any thoughts on whether we should be using `npx ember-cli` for all command examples?

File: content/ember-4-0-released.md [hunk 3]
```diff
@@ -214,18 +212,20 @@ Want to help make Ember 4 a success? Here are some things you can do!
...
+release was a wide-ranging effort with over 50 API removals in
+the `ember-source` package alone, and required coordination of timelines and
+execution across all Ember's core libraries and teams. Ember is a project
+maintained largely by volunteers, and this release is the result of sustained
+contribution across many codebases, timezones, and interests.
 
-And thank you, Ember developers, for using these tools to build awesome
-things. We appreciated your patience with the delays in our original
-release timeline.
+We deeply appreciate the support of our contributors and user community. Thank
```
Comment (jenweber) [inline content/ember-4-0-released.md:261]: @mixonic One thing we chatted about in the framework meeting is that somewhere, we should acknowledge the timeline shifts. Any idea of where that should go?

File: content/ember-4-0-released.md [hunk 4]
```diff
@@ -2,39 +2,48 @@
...
+Ember CLI. This major version bump focuses
 the framework by removing long-deprecated APIs and support for legacy platforms.
-With these changes, Ember Octane becomes the norm!
-As planned, this release introduces no new features.
+
+Although Ember "Octane" APIs have been the default for new applications since
+Ember 3.15, the framework has continued to support "Classic" framework features
+in accordance with our semantic versioning commitment. Ember 4.0 takes a step
+forward and drops already-deprecated classic APIs, however the foundational `EmberComponent`
+and `EmberObject`/`computed` APIs are not removed in this release.
```
Comment (jenweber) [inline content/ember-4-0-released.md]: @mixonic this is perfect, thank you!

File: content/ember-4-0-released.md (hunk 3 above)
Comment (mixonic) [reply content/ember-4-0-released.md:261]: We did in an update to the road to 4.0 blog post. I'm not very interested in saying "thanks for your patience", I'd rather direct a closing paragraph with thanks at the many volunteers who worked to make this happen at all.

Maybe it could be mentioned up front, just as a "originally 4.0 was expected on X date"? We can discuss more on Discord as well. I don't want this to be an apology or something though. Lets just be proud to celebrate the excellent work we've done.

File: content/ember-4-0-released.md (hunk 2 above)
Comment (mixonic) [reply content/ember-4-0-released.md]: IMO npx makes sense for installation, but traditional use of the 'ember' command should be the norm after project setup.

File: content/ember-4-0-released.md (hunk 1 above)
Comment (mixonic) [reply content/ember-4-0-released.md]: We should be careful to keep the release post focused on the release. It is dense enough as is, and imo not the place to argue or advocate for release/versioning strategy in depth.

**Comment by mixonic** [comment]
FYI I have force pushed this branch after rebase.

File: content/ember-4-0-released.md [hunk 5]
```diff
@@ -0,0 +1,231 @@
...
+
+This blog post will help you understand how to upgrade and navigate any breaking changes.
+
+Additionally, today the 4.1 beta cycle begins for these same projects. Ember's
```
Comment (kategengler) [inline content/ember-4-0-released.md]: I'd drop this whole paragraph. We don't always have all the betas out by the time of the release post and we usually consider the cycle starting when Ember.js itself branches. The info also seems superfluous to the post.

File: content/ember-4-0-released.md [hunk 6]
```diff
@@ -0,0 +1,231 @@
...
+   `npx ember-cli-update --to 3.28`. More details about how to upgrade your Ember
+   app/addon are in the
+   [general upgrade guide](https://cli.emberjs.com/release/basic-use/upgrading/#upgradinganemberappitself).
+2. Resolve all deprecation warnings. Deprecated APIs are removed in Ember 4. You may need to upgrade some
+   of your dependencies if they are using deprecated APIs. Applications that need to upgrade through several versions may want to consider
+   the
+   [ember-cli-deprecation-workflow](https://github.com/mixonic/ember-cli-deprecation-workflow)
+   addon to isolate individual deprecations.
+3. Make sure your app builds successfully.
+4. Install `ember-auto-import` if your app does not already have it:
```
Comment (kategengler) [inline content/ember-4-0-released.md]: I'm not sure this is sufficient to install `ember-auto-import` v2, as it requires webpack also be installed. The README suggests `npm install --save-dev ember-auto-import webpack` Maybe we should point to the `ember-auto-import` README?

File: content/ember-4-0-released.md [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+and better compatibility with modern build and analysis tooling (think [Webpack](https://webpack.js.org/)).
+Later in Ember's 4.x release cycle, we intend to release framework libraries as
+v2 addons to take advantage of those improvements.
+
+## Changes in Ember.js 4.0
+
+Ember.js is the core of the Ember framework. It provides routing, rendering, and
+dependency injection features.
+
+Ember.js 4.0 introduces no new public API or deprecations. Instead, it is
```
Comment (kategengler) [inline content/ember-4-0-released.md]: This isn't true -- there's at least one new deprecation for `assign`

File: content/ember-4-0-released.md [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+the framework by removing long-deprecated APIs and support for legacy platforms.
+
+Although [Ember "Octane"](https://emberjs.com/editions/octane/) APIs have been
+the default for new applications since Ember 3.15, the framework has continued
+to support "Classic" framework features in accordance with our semantic
+versioning commitment. Ember 4.0 takes a step forward and drops
+already-deprecated classic APIs, however the foundational `EmberComponent` and
+`EmberObject`/`computed` APIs are not removed in this release.
+
+This blog post will help you understand how to upgrade and navigate any breaking
```
Comment (ijlee2) [inline content/ember-4-0-released.md]: I think this sentence would go better at the end of the opening paragraph `Today the Ember project ... support for legacy platforms."` Currently, the next paragraph, which begins with `Additionally`, doesn't sound right due to this sentence.

File: content/ember-4-0-released.md [hunk 9]
```diff
@@ -0,0 +1,244 @@
...
+they should not introduce new features.
+
+This strategy aims to give Ember developers the time and space to upgrade
+dependencies and navigate any breaking changes, without needing to learn lots of
+new things at the same time. In many libraries, including Ember, a major
+release means that there are breaking changes. And while it would be exciting
+for major releases to be splashy and full of new features, we think it is more
+important to reduce risks for our developers and the users that depend on them.
+
+Our hope is that that library maintainers across the JavaScript ecosystem
```
Comment (ijlee2) [inline content/ember-4-0-released.md]:
```suggestion
Our hope is that library maintainers across the JavaScript ecosystem
```

File: content/ember-4-0-released.md [hunk 10]
```diff
@@ -0,0 +1,244 @@
...
+and you have `v2` of `ember-auto-import` installed, you should be able to
+upgrade to Ember 4 with no additional changes.
+
+Follow these steps in order:
+
+1. Upgrade your project to the latest patch version of `v3.28`.
+   Many developers can do this by running
+   `npx ember-cli-update --to 3.28`. More details about how to upgrade your Ember
+   app/addon are in the
+   [general upgrade guide](https://cli.emberjs.com/release/basic-use/upgrading/#upgradinganemberappitself).
```
Comment (ijlee2) [inline content/ember-4-0-released.md]: I think we could reverse the order to make the sentence sound more active. (similarly for lines 104-106)

```suggestion
   `npx ember-cli-update --to 3.28`. The [general upgrade guide](https://cli.emberjs.com/release/basic-use/upgrading/#upgradinganemberappitself) has more details about how to upgrade your Ember app/addon.
```

File: content/ember-4-0-released.md [hunk 11]
```diff
@@ -0,0 +1,244 @@
...
+2. Resolve all deprecation warnings. Deprecated APIs are removed in Ember 4. You
+   may need to upgrade some of your dependencies if they are using deprecated
+   APIs. See the [Ember Deprecation Guide](https://deprecations.emberjs.com/)
+   for more details about specific deprecations and how to resolve them.
+   Applications that need to upgrade through several versions may want to
+   consider the
+   [ember-cli-deprecation-workflow](https://github.com/mixonic/ember-cli-deprecation-workflow)
+   addon to isolate individual deprecations.
```
Comment (ijlee2) [inline content/ember-4-0-released.md]:
```suggestion
   consider isolating individual deprecations by using
   [ember-cli-deprecation-workflow](https://github.com/mixonic/ember-cli-deprecation-workflow).
```

File: content/ember-4-0-released.md [hunk 12]
```diff
@@ -0,0 +1,244 @@
...
+3. Make sure your app builds successfully.
+4. Install `ember-auto-import` if your app does not already have it follow the
```
Comment (ijlee2) [inline content/ember-4-0-released.md]:
```suggestion
4. Install `ember-auto-import` if your app does not already have it. Follow the
```

File: content/ember-4-0-released.md [hunk 13]
```diff
@@ -0,0 +1,244 @@
...
+Ember 4.0 supports two classes of browsers: Evergreen (those on a weeks-long, auto-upgrade release cycle) and non-evergreen. This classification system allows us to create a rolling minimum version for evergreen browsers, while using a more traditional, pinned minimum version for non-evergreen browsers.
+
+Specifically, the Ember 4.x release policy includes support for Google Chrome, Mozilla Firefox, Microsoft Edge, and Apple Safari on desktop and mobile. **It does not include support for any version of Internet Explorer.**
+
+Read more about this change in the [deprecation guide](https://deprecations.emberjs.com/v3.x/#toc_3-0-browser-support-policy) and at Ember's [browser support policy page](https://emberjs.com/browser-support/).
+
+### `ember-auto-import` is now required
+
+Using Ember 4 or higher requires that the project has version `2.0.0` or later
+of `ember-auto-import` installed. Follow the "How to upgrade" steps above to
```
Comment (ijlee2) [inline content/ember-4-0-released.md]: If possible, I would try adding a hyperlink to the section above.

```md
Follow the steps in [How to upgrade](#some-id...).
```

File: content/ember-4-0-released.md [hunk 14]
```diff
@@ -0,0 +1,244 @@
...
+learn what you need to do.
+
+Why is this change required? `ember-auto-import` supports something we call the
+[v2 addon
+format](https://emberjs.github.io/rfcs/0507-embroider-v2-package-format.html).
+This specification describes how to publish an Ember CLI addon which has minimal
```
Comment (ijlee2) [inline content/ember-4-0-released.md]:
```suggestion
This specification describes how to publish an Ember CLI addon that has minimal
```

File: content/ember-4-0-released.md [hunk 15]
```diff
@@ -0,0 +1,244 @@
...
+and the removal of previously deprecated public API from the 3.x cycle.
+
+Ember 4.0 does _not_ remove the `EmberComponent` API or the core parts of the
+`EmberObject` system. These APIs are widely used, even after the release of
+Octane, by existing application and addon code.
+
+### APIs Removed in 4.0
+
+Below we've listed some of the most significant API removals in Ember.js 4.0.
+and for an full list of removals, see the [Ember.js 3.x deprecation
```
Comment (ijlee2) [inline content/ember-4-0-released.md]:
```suggestion
For a complete list of removals, see the [Ember.js 3.x deprecation
```

File: content/ember-4-0-released.md [hunk 16]
```diff
@@ -0,0 +1,244 @@
...
+  guide](https://deprecations.emberjs.com/v3.x)
+
+Many of these removed APIs date back to Ember 1.x, and are rarely used now (or should be rarely used).
+
+For more details on the changes in Ember.js 4.0, please review the
+[Ember.js 4.0.0 release page](https://github.com/emberjs/ember.js/releases/tag/v4.0.0).
+
+### Deprecations added in 4.0
+
+Public API deprecations added in Ember 4.0 target removal in Ember 5.0. One
```
Comment (ijlee2) [inline content/ember-4-0-released.md]: I found the first sentence hard to parse and understand. If I understood the gist, I think the following will be easier to read.

```suggestion
Public API, marked as deprecated in Ember 4.0, will be removed in Ember 5.0. One
```

File: content/ember-4-0-released.md [hunk 17]
```diff
@@ -0,0 +1,244 @@
...
+such deprecation was introduced:
+
+- Use of `Ember.assign` is deprecated. You should replace any calls to `Ember.assign` with `Object.assign` or use the object spread operator. See the [deprecation guide](https://deprecations.emberjs.com/v4.x#toc_ember-polyfills-deprecate-assign) for examples.
+
+## Changes in Ember Data 4.0
+
+---
+
```
Comment (ijlee2) [inline content/ember-4-0-released.md]: Not sure why `---` appears here and a couple of more times below.

```suggestion
```

File: content/ember-4-0-released.md [hunk 18]
```diff
@@ -0,0 +1,244 @@
...
+Ember Data is the official data persistence library for Ember.js applications.
+
+This release removes APIs that were [deprecated in the 3.x cycle](https://deprecations.emberjs.com/ember-data/v3.x). They include:
```
Comment (ijlee2) [inline content/ember-4-0-released.md]:
```suggestion
This release removes APIs that were [deprecated in the 3.x cycle](https://deprecations.emberjs.com/ember-data/v3.x). Examples include:
```

File: content/ember-4-0-released.md [hunk 19]
```diff
@@ -0,0 +1,244 @@
...
+- The [Evented API is removed](https://deprecations.emberjs.com/ember-data/v3.x#toc_evented-api-usage)
+- ...and other APIs listed in the [deprecation guide](https://deprecations.emberjs.com/ember-data/v3.x)
+
+For more details on the changes in Ember Data 4.0, please review the
+[Ember Data 4.0.0 release page](https://github.com/emberjs/data/releases/tag/v4.0.0).
+
+## Changes in Ember CLI 4.0
+
+---
+
```
Comment (ijlee2) [inline content/ember-4-0-released.md]:
```suggestion
```

File: content/ember-4-0-released.md [hunk 20]
```diff
@@ -0,0 +1,244 @@
...
+Ember CLI is the command line interface for managing and packaging Ember.js
+applications.
+
+There are no deprecations, removals, or new features in Ember CLI v4.
+
+For more details on the changes in Ember CLI 4.0 and detailed upgrade
+instructions, please review the [Ember CLI 4.0.0 release page](https://github.com/ember-cli/ember-cli/releases/tag/v4.0.0).
+
+---
+
```
Comment (ijlee2) [inline content/ember-4-0-released.md]:
```suggestion
```

File: content/ember-4-0-released.md [hunk 21]
```diff
@@ -0,0 +1,244 @@
...
+For applications using LTS releases, upgrading directly to 4.0 is not
+encouraged. Best practice for those applications is to upgrade to Ember 3.28 LTS
+as soon as possible, then wait for the first LTS of the 4.0 series to be
+released. Ember 4.4 will be the first candidate.
+
+## Major Releases in Ember
+
+Why are there no new features in this release?
+This has been the tradition for all major Ember releases since v2.
```
Comment (ijlee2) [inline content/ember-4-0-released.md]: A nitpick. The word `all` may sound weird because there were only 2 major versions before (not several). I think it's okay to leave the word out.

```suggestion
This has been the tradition for major Ember releases since v2.
```

**Comment by mixonic** [comment]
Thank you @ijlee2! As always I really, really appreciate your review. :bow:

**Comment by jenweber** [comment]
@mixonic I just pushed a commit with a few changes:

- I shortened the section on major releases. We will offer more detailed info in the release page of the website someday.
- We added `npx` back in front of ember serve. The command will not succeed otherwise, unless someone already has ember cli globally installed.
- We added a note about making sure people are on the latest npm before trying to generate a project. This is to avoid the bug described here: https://github.com/ember-cli/ember-cli/issues/9711

File: content/ember-4-0-released.md [hunk 22]
```diff
@@ -0,0 +1,240 @@
...
+- `Copyable` mixin is removed in favor of the [ember-copy addon](https://github.com/emberjs/ember-copy). [Guide here](https://deprecations.emberjs.com/v3.x/#toc_ember-runtime-deprecate-copy-copyable).
+- `sendAction` is removed in favor of calling closure actions like any other callback. [Guide here](https://deprecations.emberjs.com/v3.x/#toc_ember-component-send-action).
+- `willTransition` and `didTransition` are removed in favor of router service events. [Guide here](https://deprecations.emberjs.com/v3.x/#toc_deprecate-router-events).
+- Computed Property `volatile()` calls are removed in favor of native getters. [Guide here](https://deprecations.emberjs.com/v3.x/#toc_computed-property-volatile).
+- `this.$()` and other jQuery APIs are deprecated in favor of native browser equivalents. [Guide here](https://deprecations.emberjs.com/v3.x/#toc_jquery-apis). An optional feature which restored this and other jQuery-specific features is also removed. [Guide here](https://deprecations.emberjs.com/v3.x/#toc_optional-feature-jquery-integration).
```
Comment (jenweber) [inline content/ember-4-0-released.md]: Is this deprecated, or removed in v4? This bullet point should be updated if it has been removed.

File: content/ember-4-0-released.md [hunk 23]
```diff
@@ -61,9 +49,12 @@ If you're interested in trying Ember for the
 first time today, get started by running:
 
 ```bash
-npx ember-cli new my-project
+# Make sure you are the latest npm version
```
Comment (kategengler) [inline content/ember-4-0-released.md]:
```suggestion
# Make sure you are on the latest npm version, 8.x
```

File: content/ember-4-0-released.md [hunk 24]
```diff
@@ -0,0 +1,233 @@
...
+  You should [specify adapter types explicitly](https://deprecations.emberjs.com/ember-data/v3.x#toc_ember-data-default-adapter).
+- Similarly, `adapter.defaultSerializer` and support for relying on fallback behavior for the
+  serializer type [is removed](https://deprecations.emberjs.com/ember-data/v3.x#toc_ember-data-default-serializers)
+- The [Evented API is removed](https://deprecations.emberjs.com/ember-data/v3.x#toc_evented-api-usage)
+- ...and other APIs listed in the [deprecation guide](https://deprecations.emberjs.com/ember-data/v3.x)
+
+For more details on the changes in Ember Data 4.0, please review the
+[Ember Data 4.0.0 release page](https://github.com/emberjs/data/releases/tag/v4.0.0).
+
+## Changes in Ember CLI 4.0
```
Comment (bertdeblock) [inline content/ember-4-0-released.md:208]: Maybe (some of) these might be worth including in the Ember CLI section?

- https://github.com/ember-cli/ember-cli/pull/9563
- https://github.com/ember-cli/ember-cli/pull/9579
- https://github.com/ember-cli/ember-cli/pull/9595

File: content/ember-4-0-released.md [hunk 25]
```diff
@@ -0,0 +1,233 @@
...
+
+Are you stuck on something? Do you have questions?
+Visit one of our [community chat groups](https://emberjs.com/community).
+
+## What next?
+
+Want to help make Ember 4 a success? Here are some things you can do!
+
+- Blog about your experience
+- If someone helps you with a question, consider positing both the question
```
Comment (bertdeblock) [inline content/ember-4-0-released.md]:
```suggestion
- If someone helps you with a question, consider posting both the question
```

Comment (bertdeblock) [reply content/ember-4-0-released.md]: Oh wow, it seems `positing` is a word. Sorry if this suggestion is incorrect! I thought it was a typo 😬

File: content/ember-4-0-released.md [hunk 26]
```diff
@@ -0,0 +1,243 @@
...
+
+For more details on the changes in Ember.js 4.0, please review the
+[Ember.js 4.0.0 release page](https://github.com/emberjs/ember.js/releases/tag/v4.0.0).
+
+### Deprecations added in 4.0
+
+The following public API, marked as deprecated in Ember 4.0, will be removed in
+Ember 5.0:
+
+- Use of `Ember.assign` is deprecated. You should replace any calls to `Ember.assign` with `Object.assign` or use the object spread operator. See the [deprecation guide](https://deprecations.emberjs.com/v4.x#toc_ember-polyfills-deprecate-assign) for examples.
```
Comment (MelSumner) [inline content/ember-4-0-released.md]:
```suggestion
- Use of `Ember.assign` is deprecated. You should replace any calls to `Ember.assign` with `Object.assign` or use the [object spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax). See the [deprecation guide](https://deprecations.emberjs.com/v4.x#toc_ember-polyfills-deprecate-assign) for examples.
```

File: content/ember-4-0-released.md (hunk 24 above)
Comment (MelSumner) [reply content/ember-4-0-released.md:208]: hmmm. Major versions are supposed to only be about removing deprecated APIs not adding new things. Were these really added in 4.0?

File: content/ember-4-0-released.md (hunk 25 above)
Comment (MelSumner) [reply content/ember-4-0-released.md]: I think it's the incorrect use of the word `positing` anyway, so your suggestion should still be considered.

File: content/ember-4-0-released.md [hunk 27]
```diff
@@ -0,0 +1,243 @@
...
+Visit one of our [community chat groups](https://emberjs.com/community).
+
+## What next?
+
+Want to help make Ember 4 a success? Here are some things you can do!
+
+- Blog about your experience
+- If someone helps you with a question, consider positing both the question
+  and answer to a public place like Stack Overflow or the Ember Discourse forums,
+  so that others can learn from it
```
Comment (MelSumner) [inline content/ember-4-0-released.md]:
```suggestion
  so that others can learn from it.
```

File: content/ember-4-0-released.md [hunk 28]
```diff
@@ -0,0 +1,243 @@
...
+- Help your favorite addons to upgrade to Ember 4
```
Comment (MelSumner) [inline content/ember-4-0-released.md]:
```suggestion
- Help your favorite addons to upgrade to Ember 4!
```

File: content/ember-4-0-released.md [hunk 29]
```diff
@@ -0,0 +1,243 @@
...
+## Need help?
+
+Are you stuck on something? Do you have questions?
+Visit one of our [community chat groups](https://emberjs.com/community).
+
+## What next?
+
+Want to help make Ember 4 a success? Here are some things you can do!
+
+- Blog about your experience
```
Comment (MelSumner) [inline content/ember-4-0-released.md]:
```suggestion
- Blog about your experience.
```

File: content/ember-4-0-released.md (hunk 24 above)
Comment (bertdeblock) [reply content/ember-4-0-released.md:208]: Yes => https://github.com/ember-cli/ember-cli/blob/master/CHANGELOG.md#v400.

**Review by MelSumner — APPROVED** [review]
I've suggested a few changes but they are non-blockers. It would help to have some clarity on whether or not ember-cli really did add new things in 4.0.

File: content/ember-4-0-released.md (hunk 1 above)
Comment (MelSumner) [reply content/ember-4-0-released.md]: I agree with @mixonic - @NullVoxPopuli if you want to DM me in Discord, I'll help you coordinate this. I'm going to resolve this conversation for now.

File: content/ember-4-0-released.md (hunk 22 above)
Comment (MelSumner) [reply content/ember-4-0-released.md]: Looking at [this PR](https://github.com/emberjs/rfc-tracking/issues/3) I don't think the work was completed, but maybe @rwjblue or @simonihmig can check the list to see if the work was completed and the list just wasn't updated.

Comment (simonihmig) [reply content/ember-4-0-released.md]: The list wasn't updated, will leave a comment there. The work is done. And yes, they have been *removed*!

File: content/ember-4-0-released.md [hunk 30]
```diff
@@ -0,0 +1,243 @@
...
+If you're interested in trying Ember for the
+first time today, get started by running:
+
+```bash
+# Make sure you are on the latest npm version, 8.x
+npm --version
+npx ember-cli@latest new my-project
+cd my-project
+npx ember serve
+# Then visit http://localhost:4200
```
Comment (mixonic) [inline content/ember-4-0-released.md:57]: Please confirm these exact commands work as stated before merge (or someone comment here that they have done so).

Comment (bertdeblock) [reply content/ember-4-0-released.md:57]: In this example, `npx ember serve` will work, because `ember` is the name of the executable within `/node_modules/.bin`. It's the first time I've seen it like this in any documentation though. Running e.g. `npx ember new` somewhere else will not work as it must be `npx ember-cli new` instead. After creating a new app, Ember CLI suggests to run `yarn start` or `npm start`. So that could be suggested as well instead of `npx ember serve`? And the docs use `ember server`.

Comment (mixonic) [reply content/ember-4-0-released.md:57]: Yeah I really don't like suggesting `npm ember s` here when it is only using the local path. `npx`, to me, always implies running some thing from the package `ember`. In this case it is running something from "locally installed ember-cli", a usage which I've never seen.

I don't know why we're introducing a new style of usage here and now. I'd like top revert it, again. @jenweber I think you brought it back after an earlier reversion, or had comments elsewhere here.

File: content/ember-4-0-released.md (hunk 22 above)
Comment (mixonic) [reply content/ember-4-0-released.md]: I've updated the text here appropriately. I expect it was left over from a copy-paste job.

File: content/ember-4-0-released.md (hunk 24 above)
Comment (mixonic) [reply content/ember-4-0-released.md:208]: I've updated the post to call out the CI provider change (also GitHub Actions is now the default in 4.0) and the exclude/include change.

I left out the pnpm PR as I'm not sure how much that is a step toward pnpm support vs "pnpm just works well everywhere now". I think the latter would be worth a call-out, but highlighting an incremental step might be a bit confusing.

I asked @Turbo87 for feedback on Discord. He could always update this in a followup.

File: content/ember-4-0-released.md (hunk 30 above)
Comment (mixonic) [reply content/ember-4-0-released.md:57]: Prior discussion:

* https://github.com/ember-learn/ember-blog/pull/1036#discussion_r753834945
* https://github.com/ember-learn/ember-blog/pull/1036#issuecomment-990034431

I think we agree not using `npx ember s` is the norm. I'm not clear on why @jenweber said `ember s` wouldn't work without the global installed. I thought it would be found from `node_modules/.bin` as well. If it isn't, then what we really want is `yarn run ember s`, etc.

Actually, I think `npm start` is the right thing to do here. I'm going to update to that.

**Comment by mixonic** [comment]
I believe this is actually ready to roll. It has been rebased. Date is set to 2021-12-20.
